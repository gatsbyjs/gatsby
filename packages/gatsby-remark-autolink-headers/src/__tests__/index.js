const Remark = require(`remark`)
const toString = require(`mdast-util-to-string`)
const visit = require(`unist-util-visit`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

describe(`gatsby-remark-autolink-headers`, () => {
  it(`adds id to a markdown header`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const transformed = plugin({ markdownAST }, {})

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds ids to each markdown header`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)

    const transformed = plugin({ markdownAST }, {})

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })

  it(`adds id to a markdown header with custom svg icon`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)
    const icon = `<svg width="400" height="110"><rect width="300" height="100" /></svg>`

    const transformed = plugin({ markdownAST }, { icon })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds ids to each markdown header with custom svg icon`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)
    const icon = `<svg width="400" height="110"><rect width="300" height="100" /></svg>`

    const transformed = plugin({ markdownAST }, { icon })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })

  it(`adds id to a markdown header with custom class`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)
    const className = `custom-class`

    const transformed = plugin({ markdownAST }, { className })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds ids to each markdown header with custom class`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)
    const className = `custom-class`

    const transformed = plugin({ markdownAST }, { className })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })

  it(`adds id to a markdown header with no icon`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)
    const icon = false

    const transformed = plugin({ markdownAST }, { icon })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds ids to each markdown header with no icon`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)
    const icon = false

    const transformed = plugin({ markdownAST }, { icon })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })

  it(`maintain case of markdown header for id`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)
    const maintainCase = true

    const transformed = plugin({ markdownAST }, { maintainCase })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`keeps accents by default`, () => {
    const markdownAST = remark.parse(`
# Héading One

## Héading Two

### Héading Three
    `)
    const removeAccents = false

    const transformed = plugin({ markdownAST }, { removeAccents })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
      expect(node.data.id).toEqual(expect.stringMatching(/^héading/))
    })
  })

  it(`remove accents if removeAccents is passed`, () => {
    const markdownAST = remark.parse(`
# Héading One

## Héading Two

### Héading Three
    `)
    const removeAccents = true

    const transformed = plugin({ markdownAST }, { removeAccents })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
      expect(node.data.id).toEqual(expect.stringMatching(/^heading/))
    })
  })

  it(`extracts custom id`, () => {
    const markdownAST = remark.parse(`
# Heading One {#custom_h1}

## Heading Two {#custom-heading-two}

# With *Bold* {#custom-withbold}

# Not italic {#not_italic_here}

# No custom ID

# {#id-only}

# {#text-after} custom ID
    `)
    const enableCustomId = true

    const transformed = plugin({ markdownAST }, { enableCustomId })

    const headers = []
    visit(transformed, `heading`, node => {
      headers.push({ text: toString(node), id: node.data.id })
    })
    expect(headers).toEqual([
      {
        id: `custom_h1`,
        text: `Heading One`,
      },
      {
        id: `custom-heading-two`,
        text: `Heading Two`,
      },
      {
        id: `custom-withbold`,
        text: `With Bold`,
      },
      {
        id: `not_italic_here`,
        text: `Not italic`,
      },
      {
        id: `no-custom-id`,
        text: `No custom ID`,
      },
      {
        id: `id-only`,
        text: `{#id-only}`,
      },
      {
        id: `text-after-custom-id`,
        text: `{#text-after} custom ID`,
      },
    ])
  })

  it(`adds places anchor after header when isIconAfterHeader prop is passed`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const isIconAfterHeader = true
    const transformed = plugin({ markdownAST }, { isIconAfterHeader })

    visit(transformed, `heading`, node => {
      expect(node.data.hProperties.style).toInclude(`position:relative`)
      expect(node.children).toHaveLength(2)
      expect(node.children[1].data.hProperties.class).toContain(`after`)

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds scroll-margin-top when offsetY prop is passed`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const offsetY = 32
    const transformed = plugin({ markdownAST }, { offsetY })

    visit(transformed, `heading`, node => {
      expect(node.data.hProperties.style).toInclude(
        `scroll-margin-top:${offsetY}px`
      )

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds anchor after header and scroll-margin-top when isIconAfterHeader and offsetY props are passed`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const isIconAfterHeader = true
    const offsetY = 32
    const transformed = plugin({ markdownAST }, { isIconAfterHeader, offsetY })

    visit(transformed, `heading`, node => {
      expect(node.data.hProperties.style).toInclude(`position:relative`)
      expect(node.children).toHaveLength(2)
      expect(node.children[1].data.hProperties.class).toContain(`after`)

      expect(node.data.hProperties.style).toInclude(
        `scroll-margin-top:${offsetY}px`
      )

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds id to a markdown heading when elements prop is passed with matching heading type`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const transformed = plugin({ markdownAST }, { elements: [`h1`] })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`does not add data to a markdown heading when elements prop is passed with no matching heading type`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const transformed = plugin({ markdownAST }, { elements: [`h2`] })

    visit(transformed, `heading`, node => {
      expect(node.data).not.toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`does not add data to a markdown heading with custom id when elements prop is passed with no matching heading type`, () => {
    const markdownAST = remark.parse(`# Heading Uno {#custom_h1}`)

    const transformed = plugin({ markdownAST }, { elements: [`h2`] })

    visit(transformed, `heading`, node => {
      expect(node.data).not.toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`only adds ids to markdown headings whose heading type is included in the passed elements prop`, () => {
    const markdownAST = remark.parse(`
    # Heading One

    ## Heading Two

    ### Heading Three
        `)

    const transformed = plugin({ markdownAST }, { elements: [`h2`] })

    visit(transformed, `heading`, node => {
      if (node.depth === 2) {
        expect(node.data.id).toBeDefined()
      } else {
        expect(node.data).not.toBeDefined()
      }
    })
  })

  it(`does not add data to markdown headings when an empty array elements prop is passed`, () => {
    const markdownAST = remark.parse(`
    # Heading One

    ## Heading Two

    ### Heading Three
        `)

    const transformed = plugin({ markdownAST }, { elements: [] })

    visit(transformed, `heading`, node => {
      expect(node.data).not.toBeDefined()
    })
  })

  it(`allows all six heading depths to be passed in the elements prop`, () => {
    const markdownAST = remark.parse(`
    # Heading One

    ## Heading Two

    ### Heading Three

    #### Heading Four

    ##### Heading Five

    ###### Heading Six
        `)

    const transformed = plugin(
      { markdownAST },
      { elements: [`h1`, `h2`, `h3`, `h4`, `h5`, `h6`] }
    )

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })
})
