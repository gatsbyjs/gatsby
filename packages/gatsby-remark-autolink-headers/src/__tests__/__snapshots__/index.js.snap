// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-autolink-headers adds anchor after header and scroll-margin-top when isIconAfterHeader and offsetY props are passed 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor after",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;scroll-margin-top:32px;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds id to a markdown header 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds id to a markdown header with custom class 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "custom-class before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds id to a markdown header with custom svg icon 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg width=\\"400\\" height=\\"110\\"><rect width=\\"300\\" height=\\"100\\" /></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds id to a markdown header with no icon 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds id to a markdown heading when elements prop is passed with matching heading type 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds places anchor after header when isIconAfterHeader prop is passed 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor after",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers adds scroll-margin-top when offsetY prop is passed 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "heading uno permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
      "style": "position:relative;scroll-margin-top:32px;",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers does not add data to a markdown heading when elements prop is passed with no matching heading type 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers does not add data to a markdown heading with custom id when elements prop is passed with no matching heading type 1`] = `
Object {
  "children": Array [
    Object {
      "position": Object {
        "end": Object {
          "column": 27,
          "line": 1,
          "offset": 26,
        },
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno {#custom_h1}",
    },
  ],
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 27,
      "line": 1,
      "offset": 26,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers maintain case of markdown header for id 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "Heading One permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#Heading-One",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 14,
          "line": 2,
          "offset": 14,
        },
        "start": Object {
          "column": 3,
          "line": 2,
          "offset": 3,
        },
      },
      "type": "text",
      "value": "Heading One",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "Heading-One",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "Heading-One",
    },
    "id": "Heading-One",
  },
  "depth": 1,
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 2,
      "offset": 14,
    },
    "start": Object {
      "column": 1,
      "line": 2,
      "offset": 1,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers maintain case of markdown header for id 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "Heading Two permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#Heading-Two",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 15,
          "line": 4,
          "offset": 30,
        },
        "start": Object {
          "column": 4,
          "line": 4,
          "offset": 19,
        },
      },
      "type": "text",
      "value": "Heading Two",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "Heading-Two",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "Heading-Two",
    },
    "id": "Heading-Two",
  },
  "depth": 2,
  "position": Object {
    "end": Object {
      "column": 15,
      "line": 4,
      "offset": 30,
    },
    "start": Object {
      "column": 1,
      "line": 4,
      "offset": 16,
    },
  },
  "type": "heading",
}
`;

exports[`gatsby-remark-autolink-headers maintain case of markdown header for id 3`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-label": "Heading Three permalink",
          "class": "anchor before",
        },
      },
      "title": null,
      "type": "link",
      "url": "#Heading-Three",
    },
    Object {
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 6,
          "offset": 49,
        },
        "start": Object {
          "column": 5,
          "line": 6,
          "offset": 36,
        },
      },
      "type": "text",
      "value": "Heading Three",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "Heading-Three",
      "style": "position:relative;",
    },
    "htmlAttributes": Object {
      "id": "Heading-Three",
    },
    "id": "Heading-Three",
  },
  "depth": 3,
  "position": Object {
    "end": Object {
      "column": 18,
      "line": 6,
      "offset": 49,
    },
    "start": Object {
      "column": 1,
      "line": 6,
      "offset": 32,
    },
  },
  "type": "heading",
}
`;
